cpp/mrc/src/public/coroutines/task_container.cpp

```diff
@@ -155,11 +175,19 @@ auto TaskContainer::make_cleanup_task(Task<void> user_task, task_position_t pos)
         LOG(ERROR) << "coro::task_container user_task had unhandle exception, not derived from std::exception.\n";
     }
 
-    std::scoped_lock lk{m_mutex};
-    m_tasks_to_delete.push_back(pos);
-    // This has to be done within scope lock to make sure this coroutine task completes before the
-    // task container object destructs -- if it was waiting on .empty() to become true.
-    m_size.fetch_sub(1, std::memory_order::relaxed);
+    {
+        std::scoped_lock lk{m_mutex};
+        m_tasks_to_delete.push_back(pos);
+        // This has to be done within scope lock to make sure this coroutine task completes before the
+        // task container object destructs -- if it was waiting on .empty() to become true.
+        m_size.fetch_sub(1, std::memory_order::relaxed);
+    }
+
+    if (not m_next_tasks.empty())
+    {
+        start_next_task();
+    }
```

Comment:
Why is `m_next_tasks` not included in the scoped lock? Im concerned we might have a race condition. This class guards `m_next_tasks`, `m_tasks_to_delete` and `m_size` with a lock in other places of the code. However, we are reading the contents of `m_next_tasks` without the lock. Isnt is possible that a completing task could decrement `m_next_tasks` and trigger a race condition?

Reply:
totally possible. I'll fix it.